### DISCUSSION TITLE
Can I have a group all non-major dependencies that aren't pinned config (Variant of `group:allNonMajor`) ?

### DISCUSSION THREAD

**Discussion by segevfiner** [renovatebot/renovate#25531] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

A variant of `group:allNonMajor` that doesn't group any dependency that was already pinned in the `package.json`. As in, not specified as a range, e.g. `^1.2.3`.

### What you tried so far.

We previously used the lock file maintenance config to update minor and patch dependencies, which renovate used to not even show in the dashboard for pnpm. At some point they started showing in the dashboard, but they fail to create a PR with a "Updating pnpm-lock.yam is not supported" message (Or something similar), which also prevents renovate from creating security PRs that are only a minor/patch update (As in, match the existing range).

To workaround this issue, I tried to set `rangeStrategy` to `bump`, which will also match better with the default behavior of pnpm 8, but to avoid creating a PR for every patch and minor update, I wanted renovate to group them, like would have happened with the lock file maintenance PR, into a single PR. So I have enabled `group:allNonMajor`. 

But this configuration will also group pinned dependencies into the all non-major PR, which I don't want. I only want to group minor/patch updates of non-pinned dependencies. That is, ones that have a range set in the `package.json`.

How can I achieve this? 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
You could write your own package rule and utilize [matchCurrentValue](https://docs.renovatebot.com/configuration-options/#matchcurrentvalue) with a regex to match or not match values which start with a digit

**Comment by segevfiner** [comment]
So something like this:
```json
    {
      "groupName": "all npm non-major dependencies",
      "groupSlug": "all-npm-minor-patch",
      "matchManagers": ["npm"],
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "matchCurrentValue": "/^[~^]/",
  }
```
